x-pack/plugins/stack_connectors/server/connector_types/crowdstrike/crowdstrike.ts

```diff
@@ -199,13 +200,26 @@ export class CrowdstrikeConnector extends SubActionConnector<
   }
 
   protected getResponseErrorMessage(
-    error: AxiosError<{ errors: [{ message: string; code: number }] }>
+    error: AxiosError<{ errors: Array<{ message: string; code: number }> }>
   ): string {
     const errorData = error.response?.data?.errors?.[0];
     if (errorData) {
       return errorData.message;
     }
 
+    // @ts-expect-error TODO: Can't find a way to make the types work here
```

Comment:
Yeah not sure, I wasn't able to cast this properly. But this is what I get as `error`

Reply:
As Patrick said, based on how the type is defined here https://github.com/axios/axios/blob/0e4f9fa29077ebee4499facea6be1492b42e8a26/index.d.ts#L402 the generic (`<{ errors ... }>`) is for the response and not for the cause. The cause is an [instance](https://github.com/axios/axios/blob/0e4f9fa29077ebee4499facea6be1492b42e8a26/lib/core/AxiosError.js#L91) of `Error`. Do you have access to the information you are looking for in `error.response`?

Reply:
hey @cnasikas, thanks for taking a look 👍  
Unfortunately in these cases where I get 'ENOTFOUND` or `ECONNREFUSED` , there is no `error.response`. 
These errors are not coming from the Crowdstrike API, but rather from the fact that I am trying to hit an unexisting API. 
Anyway, it's a bit confusing to me too.

Reply:
I see. Based on the Node.js documentation this is a Node.js [System error](https://github.com/elastic/kibana/pull/184384). So Node.js throws a system error and Axios is putting the `Error` object to `error.cause`. As you said there is no response because you do not get a response from the Crowdstrike API.

Reply:
I would say to remove the `@ts-expect-error` and cast it to a `SystemError`. Wdyt?

Reply:
Yes, I was able to reproduce 2 kind of errors: `ENOTFOUND` or `ECONNREFUSED` - I don't remember which of these was it, but it contained an array, thus the check for length. 

I'll try to cast this as system error, thanks for the hint! :)

Reply:
You are welcome! It may cause you issues with the `cause.hostname`. If that's the case feel free to cast it to your own structure.